semver.go

```diff
@@ -11,32 +12,78 @@ const (
 	numbers  string = "0123456789"
 	alphas          = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-"
 	alphanum        = alphas + numbers
+
+	// threeFourDigitVersionRegexFormat represents the strict rules for parsing 3 and 4 digit
+	// version numbers. There are six capture groups which define major, minor, patch, revision,
+	// pre-release, and build metadata. When using 3 digit versioning strict semver adherence
+	// will be utilized/
+	threeFourDigitVersionRegexFormat = `^(?P<major>0|[1-9]\d*)\.` +
```

Comment:
So I modified this from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string adding the revision and created the tolerant variant too.

Reply:
ok, so i tried it: https://go.dev/play/p/ANozC6L9sJM with surprising results:
- `^0|[1-9]\d*$` accepts "01".  not good
- `^0|([1-9]\d*)$` also accepts "01".... why?
- `^(0|[1-9]\d*)$`  does reject "01"  so it seems these are fine.

found the explanation in [a POSIX doc](https://www.boost.org/doc/libs/1_56_0/libs/regex/doc/html/boost_regex/syntax/basic_extended.html#boost_regex.syntax.basic_extended.operator_precedence).   in short: alternation '|' is the lowest precedence of all, which is good, but it's lower than anchors (^ and $).

the second pattern essentially means "starts with 0 or ends with digits", which isn't useful.  the last pattern is basically: "full string: 0 or nice number"

TL;DR: these patterns are ok because each alternation is wrapped in some group